### PR TITLE
some more transaction validation bug fixes

### LINKED ISSUE #3770: Tx Controller - gasLimit is not autofilled
Submitting a tx with no gasLimit specified, has a gasLimit of `0` when approved

Old ui now throws a validation warning,
New ui submits tx with 0 gasLimit

we should be auto setting the gasLimit to something sane, pretty sure we used to

### PR DESCRIPTION
**Pull request by frankiebee**
this PR fixes #3770 
plusss bonus fix validates the from field is not a "thing" and is a valid address

### CODE REVIEW

File: app/scripts/controllers/transactions.js [hunk 1]
```diff
@@ -215,6 +215,7 @@ module.exports = class TransactionController extends EventEmitter {
     }
     txParams.gasPrice = ethUtil.addHexPrefix(gasPrice.toString(16))
     txParams.value = txParams.value || '0x0'
+    if (txParams.to === null) delete txParams.to
```
Comment (kumavis) [inline app/scripts/controllers/transactions.js]: may need to handle the case where `txParams.to === '0x'` if we're not handling that somewhere else. if we are handling that somewhere else, we should put all this params normilization in the same place.

Comment (kumavis) [reply app/scripts/controllers/transactions.js]: ( mew was sending us `to: 0x` once upon a time )

**Review by kumavis — APPROVED** [review]

File: app/scripts/controllers/transactions.js (hunk 1 above)
Comment (frankiebee) [reply app/scripts/controllers/transactions.js]: eyk okay

Comment (frankiebee) [reply app/scripts/controllers/transactions.js]: shouldnt be an issue in this case we already remove to fields that would have that before this point

**Pull request merged**